src/tools/crossgen2/Common/JitInterface/CorInfoImpl.cs

```diff
@@ -674,16 +674,23 @@ private uint getMethodAttribsInternal(MethodDesc method)
             // Check for hardware intrinsics
             if (HardwareIntrinsicHelpers.IsHardwareIntrinsic(method))
             {
-#if !READYTORUN
                 // Do not report the get_IsSupported method as an intrinsic - RyuJIT would expand it to
                 // a constant depending on the code generation flags passed to it, but we would like to
                 // do a dynamic check instead.
                 if (!HardwareIntrinsicHelpers.IsIsSupportedMethod(method)
-                    || HardwareIntrinsicHelpers.IsKnownSupportedIntrinsicAtCompileTime(method))
+#if !READYTORUN
```

Comment:
The existing `ifdef` here should stay as is. @trylek made getCallInfo mirror the exact structure of getCallInfo on the CoreCLR side - ceeInfoGetCallInfo matches CEEInfo::getCallInfo and getCallInfo matches ZapInfo::getCallInfo, pretty much line-by-line.

So this should probably work the same as it does in crossgen to keep the structure - ceeInfoGetCallInfo is going to return CORINFO_FLG_JIT_INTRINSIC bit set, and then we strip it off in FilterHardwareIntrinsicMethodAttribs within getCallInfo.

Reply:
I have been experimenting with this feedback for a while today. I can't find the logic that stops the JIT from inlining the `IsSupported` methods, but I don't see them getting inlined either after I applied `Hmm - not sure`. So well, I am not sure, but it seems to work.

Reply:
This misunderstanding is resolved, this needs work.